tests/unit/cag/test_one_hot_encoding.py

```diff
@@ -0,0 +1,168 @@
+"""Unit tests for OneHotEncoding CAG pattern."""
+
+import re
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from sdv.cag import OneHotEncoding
+from sdv.cag._errors import PatternNotMetError
+from sdv.metadata import Metadata
+
+
+class TestOneHotEncoding:
+    def test___init___incorrect_column_name(self):
+        """Test it raises an error if column_name is not a string."""
+        # Run and Assert
+        err_msg = '`column_names` must be a list of strings.'
+        with pytest.raises(ValueError, match=err_msg):
+            OneHotEncoding(column_names=1)
+
+        with pytest.raises(ValueError, match=err_msg):
+            OneHotEncoding(column_names=['a', 1])
+
+    def test___init___incorrect_table_name(self):
+        """Test it raises an error if table_name is not a string."""
+        # Run and Assert
+        err_msg = '`table_name` must be a string or None.'
+        with pytest.raises(ValueError, match=err_msg):
+            OneHotEncoding(column_names=['a', 'b', 'c'], table_name=1)
+
+    def test___init___(self):
+        """Test it initializes correctly."""
+        # Run
+        instance = OneHotEncoding(column_names=['a', 'b', 'c'])
+
+        # Asserts
+        assert instance._column_names == ['a', 'b', 'c']
+        assert instance.table_name is None
+
+    @patch('sdv.cag.one_hot_encoding._validate_table_and_column_names')
+    def test__validate_pattern_with_metadata(self, validate_table_and_col_names_mock):
+        """Test validating the pattern with metadata."""
+        # Setup
+        instance = OneHotEncoding(column_names=['low', 'middle', 'high'])
+        metadata = Metadata.load_from_dict({
+            'tables': {
+                'table': {
+                    'columns': {
+                        'low': {'sdtype': 'numerical'},
```

Comment:
The current one-hot encoding constraint does not check sdtype, which is why this CAG is not. 
Do need to check sdtype for this CAG in the future?

Reply:
It doesn't work with non-numerical data, so I agree the check should be added in the future.